models/sessions.sql

```diff
@@ -1,11 +1,20 @@
 {{
     config(
-        materialized='view',
-        unique_key='full_session_id'
+        materialized='incremental',
```

Comment:
This is a breaking change, right? Do we need to consider dbt model versioning here?

Reply:
@sgagofs I don't think so. I'd much prefer to rely on semantic versioning and documentation while we are in the early phases here. Does rebuilding a view as an incrementally updated table constitute a breaking change? The original idea behind materializing these as views is that we'd be able to upgrade to incrementals as a minor version update.

Reply:
That's right, semantic versioning. Would recommend a major version number increment here then.

Rephrasing the question: Does changing a model materialization result in a breaking change? Yes, it can sometimes. In BigQuery, you have to drop the view first because a table and view cannot have the same name. Dropping anything, even briefly, could disrupt customer processes. And first time builds of incremental models could be like 2+ minutes where the models are unavailable.

Reply:
Yes, I believe this would be a major version bump for the reasons @sgagofs mentioned. I also that we should lean heavily on proper versioning protocols and _strongly_ recommend users of this package pin a version when importing.

Reply:
I agree that we should use semver properly and recommend version pinning, but the definition of "breaking change" is up to interpretation. I'm genuinely not sure of best practices here. Usually, a change is considered breaking when it breaks a contract in a backward-incompatible way. This is not the case here - all queries written against the model will still be valid. Nothing about materialization is listed either way in [this article](https://docs.getdbt.com/reference/resource-properties/versions#detecting-breaking-changes). [This article](https://docs.getdbt.com/guides/legacy/building-packages#6-release-your-package) says, "In particular, if new changes will cause errors for users of earlier versions of the package, be sure to use at least a minor release", suggesting that it is common practice to use minor versions, even if there could be errors.

At any rate, I think we're safe to bump only the minor version, regardless of how we choose to define "breaking change" w.r.t. dbt. [Semantic versioning doesn't require strict versioning for major version 0](https://semver.org/#spec-item-4).